File: Innovation-Challenge/史大伟-AI招聘助手/kangaroo_lib_for_kotlin/ktlib/src/main/java/com/kangaroo/ktlib/app/AppEnvironment.kt
package com.kangaroo.ktlib.app

import com.kangaroo.ktlib.app.init.sConfiger
import com.kangaroo.ktlib.app.init.sysDataStore
import com.kangaroo.ktlib.util.log.ULog
import kotlinx.coroutines.flow.first

/**
 * @author  SHI DA WEI
 * @date  2023/10/31 14:49
 */
enum class Environment {
    PRODUCT, DEV, TEST, LOCAL
}
object AppEnvironment {

    private var tenv:Environment? = null

    private lateinit var tenvs:Map<Environment,Map<String,String?>>

    private var tdefalutEnv = Environment.PRODUCT


    fun init(envs : Map<Environment,Map<String,String?>>,defalutEnv:Environment = Environment.PRODUCT){
        ULog.o(envs, tagToSysLogTag("AppEnvironment"))
        tenvs = envs
        if(sConfiger.debugStatic){
            tdefalutEnv = defalutEnv
        }
    }

    suspend fun setEnv(env:Environment){
        tenv = env
        sysDataStore.putData(SYS_ENV,env.name)
    }

    suspend fun getEnv():Environment{
        if(sConfiger.debugStatic){
            if(tenv!=null){
                return tenv!!
            }
            var name = sysDataStore.getData(SYS_ENV, tdefalutEnv.name).first()
            tenv = Environment.valueOf(name!!)
            return tenv!!
        }
        return tdefalutEnv
    }

    suspend fun getEnvConfig():Map<String, String?>?{
        return tenvs[getEnv()]
    }

    fun getEnvs():Map<Environment,Map<String,String?>>{
        return tenvs
    }

}

File: Innovation-Challenge/张秋伟-RickAndMorty角色扮演/kangaroo_flutter/base_lib/lib/app/app_environment.dart
import 'dart:io';

import 'package:base_lib/tools/log/log_manager.dart';
import 'package:base_lib/tools/store/store_manager.dart';

import 'application.dart';
import 'sys_config.dart';

enum Environment {
  PRODUCT,
  DEV,
  TEST,
  LOCAL,
}

class AppEnvironment {
  static Environment? _env;

  static late Map<Environment, Map<String, String?>> _envs;

  static void init(Map<Environment, Map<String, String?>> envs,{Environment defalutEnv = Environment.PRODUCT}) {
    LogManager.log.d(envs,tag: SysConfig.libEnvironmentTag);
    _envs = envs;
    if (Application.config.debugState) {
      _defalutEnv = defalutEnv;
    }
  }

  static set env(Environment env) {
    _env = env;
    StoreManager.store.setInt(SysConfig.sysEnv, env.index);
    // exit(0);
  }

  static Environment _defalutEnv = Environment.PRODUCT;


  static Environment get env {
    if (Application.config.debugState) {
      if (_env != null) {
        return _env!;
      }
      var index = StoreManager.store.getInt(SysConfig.sysEnv)??_defalutEnv.index;
      _env = Environment.values[index];
      return _env!;
    }
    return _defalutEnv;
  }

  static Map<String, String?>? get envConfig {
    return _envs[env];
  }

  static Map<Environment, Map<String, String?>> get envs {
    return _envs;
  }


}
